### DISCUSSION TITLE
Are BigInt literals output by the code gen in places aside from map value defaults?

### DISCUSSION THREAD

**Discussion by jcready** [timostamm/protobuf-ts#157] [Q&A]
One issue we're facing is that we're using Webpack v4 which doesn't support es2020 syntax and TypeScript won't handle BitInt literals if the tsconfig `target` is less than `es2020`. Updating to Webpack v5 is a large chunk of work and we are considering just patching the protobuf-ts plugin such that it outputs:
```ts
map[key ?? ""] = val ?? BigInt("0");
```
instead of
```ts
map[key ?? ""] = val ?? 0n;
```
Are you aware of any other place a BigInt literal would be generated?

**Comment by timostamm** [comment]
I don't think there is another place. 

Is `long_type_string` or `long_type_number` not an option for you? 

https://github.com/timostamm/protobuf-ts/blob/master/MANUAL.md#the-protoc-plugin

**Reply by jcready** [reply]
Those are certainly options, but we very much want the benefits of `BigInt`. And all of our runtime environments support it. The only hold up is Webpack v4 not understanding the `BigInt` literal syntax.

**Comment by timostamm** [comment]
I see. Did you try [plugin-syntax-bigint](https://babeljs.io/docs/en/babel-plugin-syntax-bigint.html)? Found via this [comment](https://github.com/webpack/webpack/issues/9077#issuecomment-574669949).
